Tests/Realm.Tests/Database/InstanceTests.cs

```diff
@@ -39,6 +39,23 @@ public void GetInstanceTest()
             GetRealm().Dispose();
         }
 
+        [Test]
+        public void ReadOnlyInstance_ThrowsOnRefresh()
+        {
+            var path = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "temp.realm");
```

Comment:
Let's not build deterministic paths and instead use `Guid.NewGuid()` here.

Reply:
Can you explain the reasoning for this?

Reply:
There are several:
1. We have code in our tests that modifies the base path for where we store Realm files to be the temp directory.
2. If you hardcode a path, this means that running the test twice may result in a non-clean slate run. For example, if the test crashes/throws halfway through, this would prevent us from deleting the Realm file. Then the next time you run it, you'll be using the existing file rather than a brand new one.

Reply:
Thank you. That makes totally sense.